sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -508,28 +527,41 @@ export function getGeneratedClientOptions(
       ...retryPolicies,
       ...defaults,
     ],
-    userAgentHeaderName: getUserAgentHeaderName(internalAppConfigOptions)
+    userAgentHeaderName: getUserAgentHeaderName(internalAppConfigOptions.isNodeOverride),
+    userAgent: getUserAgentPrefix(internalAppConfigOptions.userAgentOptions && internalAppConfigOptions.userAgentOptions.userAgentPrefix)    
   };
 }
 
 /**
- * @ignore
  * @internal
+ * @ignore
  */
-export function getUserAgentHeaderName(internalAppConfigOptions: InternalAppConfigurationClientOptions) {
-  const isNode = internalAppConfigOptions.isNodeOverride == null
-    ? isNodeFromCoreHttp
-    : internalAppConfigOptions.isNodeOverride;
+export function getUserAgentPrefix(userSuppliedUserAgent: string | undefined): string {
+  const appConfigDefaultUserAgent = `${packageName}/${packageVersion} ${getCoreHttpDefaultUserAgentValue()}`;
   
-  let userAgentHeaderName: string | undefined;
+  if (userSuppliedUserAgent == null) {
+    return appConfigDefaultUserAgent;
+  }
+  else {
```

Comment:
The else isn't needed  since you return in the if statement above. (I like anytime we can reduce indentation)

Reply:
Sure.